src/schemathesis/specs/openapi/_hypothesis.py

```diff
@@ -186,15 +190,17 @@ def is_yaml_parsing_issue(operation: APIOperation) -> bool:
 
 
 def _get_body_strategy(
-    parameter: OpenAPIParameter, to_strategy: Callable[[Dict[str, Any]], st.SearchStrategy], parent_schema: BaseSchema
+    parameter: OpenAPIParameter,
+    to_strategy: Callable[[Dict[str, Any], str], st.SearchStrategy],
+    parent_schema: BaseSchema,
 ) -> st.SearchStrategy:
     # The cache key relies on object ids, which means that the parameter should not be mutated
     # Note, the parent schema is not included as each parameter belong only to one schema
     if parameter in _BODY_STRATEGIES_CACHE and to_strategy in _BODY_STRATEGIES_CACHE[parameter]:
         return _BODY_STRATEGIES_CACHE[parameter][to_strategy]
     schema = parameter.as_json_schema()
     schema = parent_schema.prepare_schema(schema)
-    strategy = to_strategy(schema)
+    strategy = to_strategy(schema, "body")
```

Comment:
Should this be converted into an enum instead?

Reply:
Could you, please, elaborate?